### PR TITLE
codegen: fix DeepCopy races

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/api/issues/3019

The deepcopygen we use is from Kubernetes. Its not quite suitable for our types due to our usage of protobuf, which has limitations on dereferencing pointer types. This causes race conditions in concurrent usage of the DeepCopy function.

This is possible to trigger with real Istio (found by running Istio in a live cluster with `-race`, though our current tests happen to not trigger things I guess), as well as other usage of istio/client-go (as the reporter found).

I ran istio/istio with this change and found all tests continue to pass, and running in a live cluster no longer fails

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -100,6 +100,12 @@ rename_generated_files=\
 	find $(subst istio.io/client-go/, $(empty), $(subst $(comma), $(space), $(kube_api_packages)) $(kube_clientset_package) $(kube_listers_package) $(kube_informers_package)) \
 	-name '*.go' -and -not -name 'doc.go' -and -not -name '*.gen.go' -type f -exec sh -c 'mv "$$1" "$${1%.go}".gen.go' - '{}' \;
 
+# Kubernetes deepcopy gen directly sets values of our types. Our types our protos; it is illegal to do this for protos.
```
Comment (therealmitchconnors) [inline Makefile.core.mk:103]:
```suggestion
# Kubernetes deepcopy gen directly sets values of our types. Our types are protos; it is illegal to do this for protos.
```

File: Makefile.core.mk [hunk 2]
```diff
@@ -100,6 +100,12 @@ rename_generated_files=\
...
+# However, we don't even need this anyways -- each individual field is explicitly copied already.
+# Remove this line.
```
Comment (therealmitchconnors) [inline Makefile.core.mk:105]: Remove this line?

**Review by therealmitchconnors — COMMENTED** [review]
Minor nits, otherwise LGTM

File: Makefile.core.mk (hunk 2 above)
Comment (howardjohn) [reply Makefile.core.mk:105]: renamed to 'Remove the line doing this illegal operation' -- the intent was to describe what the sed does

**Review by therealmitchconnors — APPROVED** [review]

**Pull request merged**